pop-api/examples/fungibles/tests.rs

```diff
@@ -0,0 +1,397 @@
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+	// Mint tokens.
+	assert_ok!(mint(&mut session, ALICE, 0));
+	// No event emitted.
+	assert_eq!(last_contract_event(&session), None);
+	// Tokens were minted with the right amount.
+	assert_eq!(total_supply(&mut session), 0);
+	assert_eq!(balance_of(&mut session, ALICE), 0);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn burn_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+	// Mint tokens.
+	const AMOUNT: Balance = 12_000;
+	assert_ok!(mint(&mut session, ALICE, AMOUNT));
+	// Burn tokens.
+	assert_ok!(burn(&mut session, ALICE, 1));
+	// Successfully emit event.
+	let expected =
+		Transfer { from: Some(account_id_from_slice(ALICE.as_ref())), to: None, value: 1 }.encode();
+	assert_eq!(last_contract_event(&session).unwrap(), expected.as_slice());
+
+	assert_eq!(total_supply(&mut session), AMOUNT - 1);
+	assert_eq!(balance_of(&mut session, ALICE), AMOUNT - 1);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn burn_zero_value_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+	// Burn tokens.
+	assert_ok!(burn(&mut session, ALICE, 0));
+	// No event emitted.
+	assert_eq!(last_contract_event(&session), None);
+	// Tokens were minted with the right amount.
+	assert_eq!(total_supply(&mut session), 0);
+	assert_eq!(balance_of(&mut session, ALICE), 0);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn burn_fails_with_insufficient_balance(
+	mut session: Session,
+) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+	// Mint tokens.
+	const AMOUNT: Balance = 12_000;
+	assert_ok!(mint(&mut session, ALICE, AMOUNT));
+	// Failed with `InsufficientBalance`.
+	expect_call_reverted(
+		&mut session,
+		BURN,
+		vec![ALICE.to_string(), (AMOUNT + 1).to_string()],
+		PSP22Error::InsufficientBalance,
+	);
+	Ok(())
+}
+
+#[drink::test(sandbox = Sandbox)]
+fn transfer_works(mut session: Session) -> Result<(), Box<dyn std::error::Error>> {
+	let _ = env_logger::try_init();
+	// Deploy a new contract.
+	let contract = deploy_with_new_constructor(
+		&mut session,
+		BundleProvider::local()?,
+		TOKEN_ID,
+		TOKEN_MIN_BALANCE,
+	)?;
+	// Mint tokens.
+	const AMOUNT: Balance = 12_000;
+	const TRANSFERRED: Balance = 500;
+	assert_ok!(mint(&mut session, contract.clone(), AMOUNT));
```

Comment:
We should be using the mint api that calls into pallet assets directly, not the api of the contract. This counts for all tests